src/interfaces/token.ts

```diff
@@ -191,157 +187,76 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
+      owner: ownerAddress,
+      token: tokenAddress,
+      spender: spenderAddress,
+      amount: MaxUint256.toString()
     };
 
-    if (EthAddress === token) return allowance;
-
-    if (vaultToken === token) {
-      const tokenContract = new Contract(token, TokenAbi, this.ctx.provider.read);
-      const partnerAddress = await this.yearn.services.partner?.address;
-      const addressToApprove = (this.shouldUsePartnerService(address) && partnerAddress) || address;
-      const allowanceAmount = await tokenContract.allowance(account, addressToApprove);
-
-      return {
-        ...allowance,
-        spender: addressToApprove,
-        amount: allowanceAmount.toString()
-      };
-    }
+    if (isNativeToken(tokenAddress) || tokenAddress === spenderAddress) return allowance;
 
-    const zapProtocol = PickleJars.includes(address) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
-
-    const zapInApprovalState = await this.yearn.services.zapper.zapInApprovalState(account, token, zapProtocol);
+    const tokenContract = new Contract(tokenAddress, TokenAbi, this.ctx.provider.read);
+    const allowanceAmount = await tokenContract.allowance(ownerAddress, spenderAddress);
 
     return {
-      owner: zapInApprovalState.ownerAddress,
-      spender: zapInApprovalState.spenderAddress,
-      amount: zapInApprovalState.allowance,
-      token
+      ...allowance,
+      amount: allowanceAmount.toString()
     };
   }
 
   /**
-   * Approve vault to spend a token on zapIn or direct deposit
-   * @param vault
-   * @param token
+   * Approve the token amount that spender is allowed to spend on behalf of owner
+   * @param ownerAddress
+   * @param tokenAddress
+   * @param spenderAddress
    * @param amount
-   * @param account
-   * @returns transaction
+   * @param overrides
+   * @returns TokenAllowance
    */
-  async approveDeposit(
-    address: Address,
-    vaultToken: Address,
-    token: Address,
+  async approve(
+    ownerAddress: Address,
+    tokenAddress: Address,
+    spenderAddress: Address,
     amount: Integer,
-    account: Address
-  ): Promise<TransactionResponse | boolean> {
-    // If Ether is being sent, no need for approval
-    if (EthAddress === token) return true;
-
-    const signer = this.ctx.provider.write.getSigner(account);
-
-    if (vaultToken === token) {
-      const tokenContract = new Contract(token, TokenAbi, signer);
-      const partnerAddress = await this.yearn.services.partner?.address;
-      const addressToApprove = (this.shouldUsePartnerService(address) && partnerAddress) || address;
-      const tx = await tokenContract.populateTransaction.approve(addressToApprove, amount);
-      return this.yearn.services.transaction.sendTransaction(tx);
-    }
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    if (isNativeToken(tokenAddress)) throw new SdkError(`Native tokens cant be approved`);
+    if (tokenAddress === spenderAddress) throw new SdkError(`Cant approve token as its spender`);
```

Comment:
I guess no need to interpolate here, unless we want to add the `tokenAddress` as well
```suggestion
    if (tokenAddress === spenderAddress) throw new SdkError("Cant approve token as its spender");
```

Reply:
added token address too https://github.com/yearn/yearn-sdk/pull/266/commits/dedff679121d87f2112f17c56784aaece55c6a89